llama_index/vector_stores/supabase.py

```diff
@@ -75,8 +75,12 @@ def client(self) -> None:
     def _to_vecs_filters(self, filters: MetadataFilters) -> Any:
         """Convert llama filters to vecs filters. $eq is the only supported operator."""
         vecs_filter = {}
+        filter_cond = "${}".format(filters.condition.value)
+        vecs_filter[filter_cond] = []
         for f in filters.legacy_filters():
```

Comment:
we are retiring legacy_filters. pls use MetadataFilters.filters. and you can set operator from filters instead of using `$eq` as fixed